custom/icds_reports/tasks.py

```diff
@@ -1343,3 +1347,97 @@ def email_location_changes(domain, old_location_blob_id, new_location_blob_id):
         '[{}] - ICDS Dashboard Location Table Changed'.format(settings.SERVER_ENVIRONMENT),
         DASHBOARD_TEAM_EMAILS, email_content,
     )
+
+
+@periodic_task(run_every=crontab(hour=22, minute=0))
+def create_reconciliation_records():
+    # Setup yesterday's data to reduce noise in case we're behind by a lot in pillows
+    UcrReconciliationStatus.setup_days_records(date.today() - timedelta(days=1))
+
+
+@task(queue='background_queue')
+def reconcile_data_not_in_ucr(reconciliation_status_pk):
+    status_record = UcrReconciliationStatus.objects.get(pk=reconciliation_status_pk)
+    number_documents_missing = 0
+
+    # republish_kafka_changes
+    for doc_id, doc_subtype, sql_modified_on in get_data_not_in_ucr(status_record):
+        number_documents_missing += 1
+        celery_task_logger.info(f'doc_id {doc_id} from {sql_modified_on} not found in UCR data sources')
+        send_change_for_ucr_reprocessing(doc_id, doc_subtype, status_record.is_form_ucr)
+
+    status_record.last_processed_date = datetime.utcnow()
+    if number_documents_missing == 0:
+        status_record.verified_date = datetime.utcnow()
+    status_record.save()
+
+    return number_documents_missing
+
+
+def send_change_for_ucr_reprocessing(doc_id, doc_subtype, is_form):
+    producer.send_change(
+        topics.FORM_SQL if is_form else topics.CASE_SQL,
+        ChangeMeta(
+            document_id=doc_id,
+            data_source_type=data_sources.SOURCE_SQL,
+            data_source_name=data_sources.FORM_SQL if is_form else data_sources.CASE_SQL,
+            document_type='XFormInstance' if is_form else 'CommCareCase',
+            document_subtype=doc_subtype,
+            domain=DASHBOARD_DOMAIN,
+            is_deletion=False,
+        )
+    )
+
+
+def get_data_not_in_ucr(status_record):
+    domain = DASHBOARD_DOMAIN
+    if status_record.is_form_ucr:
+        matching_records_for_db = _get_primary_data_for_forms(
+            status_record.db_alias, domain, status_record.day, status_record.doc_type_filter
+        )
+    else:
+        matching_records_for_db = _get_primary_data_for_cases(
+            status_record.db_alias, domain, status_record.day, status_record.doc_type_filter
+        )
+    chunk_size = 1000
+    for chunk in chunked(matching_records_for_db, chunk_size):
+        doc_ids = [val[0] for val in chunk]
+        docs_in_ucr = _get_docs_in_ucr(domain, status_record.table_id, doc_ids)
+        for doc_id, doc_subtype, sql_modified_on in chunk:
+            if doc_id not in docs_in_ucr:
+                yield (doc_id, doc_subtype, sql_modified_on.isoformat())
+
+
+def _get_docs_in_ucr(domain, table_id, doc_ids):
+    table_name = get_table_name(domain, table_id)
+    with connections[get_icds_ucr_citus_db_alias()].cursor() as cursor:
+        query = f'''
+            SELECT doc_id
+            FROM "{table_name}"
+            WHERE doc_id = ANY(%(doc_ids)s);
```

Comment:
The management command we've been using recently have also been returning `inserted_at`. I chose to remove that so that we could use `doc_id` index only scans and at the moment we are only looking into docs not being in the data source, not "stale" data.